README.md

```diff
@@ -1,2 +1,40 @@
 # java-globaltracer
-Global ThreadLocal Span for implicit propagation, delegating to another Tracer implementation.
+Global Tracer, forwarding to another Tracer implementation.
+
+This library provides two utility classes `ActiveSpanManager` and `GlobalTracer`:
+
+## ActiveSpanManager
+This class:
+ 1. Provides clients with the `ActiveSpanManger.activeSpan()` method to return the
+    _active span_.
+    If there is no active span, the method never returns a `null` refrence, but
+    a `NoopSpan` instead.
+ 2. Modifications to the _active span_ can be made through the `activate` 
+    and `deactivate` methods.
+ 3. Utility `spanAware()` methods to create 'span aware' Runnable and Callable instances
+    that run with the _active span_ from the scheduling thread.
+
+## GlobalTracer
+This class has the following purpose:
+ 1. The `GlobalTracer.tracer()` factory method returning the singleton _global tracer_.
+    If there is no global tracer, a `NoopTracer` is returned instead.
+ 2. Enrich the lifecycle of created `Span` objects through the global Tracer 
+    to update the _active span_ as they are started and finished.
+ 3. Utility `traced()` methods to create Runnable and Callable instances 
+    that run within a new Span that is _child of the active span_ form the scheduling thread.
+    An `operationName` must be provided for a new Span to be created.
```

Comment:
Is the use of `Tracer.buildSpan(null)` (null for `operationName`) considered accepted behaviour?
In that case I would also prefer the simplification you propose.